be/src/exec/pipeline/pipeline_driver_poller.cpp

```diff
@@ -109,6 +109,11 @@ void PipelineDriverPoller::run_internal() {
                         remove_blocked_driver(_local_blocked_drivers, driver_it);
                         ready_drivers.emplace_back(driver);
                     }
+                } else if (driver->need_report_exec_state()) {
+                    // If the runtime profile is enabled, the driver should be rescheduled after the timeout for triggering
+                    // the profile report prcessing.
+                    remove_blocked_driver(_local_blocked_drivers, driver_it);
+                    ready_drivers.emplace_back(driver);
```

Comment:
Yes, I don't want to add complexity of the state management. And the report will be processed at the worker, and then it will be put back to the blocked queue without changing the state.